cpp/dynamic_community_detection_module/dynamic_community_detection_module.cpp

```diff
@@ -0,0 +1,291 @@
+#include <mg_graph.hpp>
+#include <mg_utils.hpp>
+
+#include "algorithm/dynamic_community_detection.hpp"
+
+constexpr char const *kFieldNode = "node";
+constexpr char const *kFieldCommunityId = "community_id";
+
+constexpr char const *kDirected = "directed";
+constexpr char const *kWeighted = "weighted";
+constexpr char const *kSimilarityThreshold = "similarity_threshold";
+constexpr char const *kExponent = "exponent";
+constexpr char const *kMinValue = "min_value";
+constexpr char const *kWeightProperty = "weight_property";
+constexpr char const *kWSelfloop = "w_selfloop";
+constexpr char const *kMaxIterations = "max_iterations";
+constexpr char const *kMaxUpdates = "max_updates";
+
+constexpr char const *kFieldMessage = "message";
+
+constexpr char const *kCreatedVertices = "createdVertices";
+constexpr char const *kCreatedEdges = "createdEdges";
+constexpr char const *kUpdatedVertices = "updatedVertices";
+constexpr char const *kUpdatedEdges = "updatedEdges";
+constexpr char const *kDeletedVertices = "deletedVertices";
+constexpr char const *kDeletedEdges = "deletedEdges";
+
+LabelRankT::LabelRankT algorithm = LabelRankT::LabelRankT();
+bool initialized = false;
+
+auto saved_directedness = false;
+auto saved_weightedness = false;
+std::string saved_weight_property = "weight";
+double saved_default_weight = 1.0;
```

Comment:
I don't exactly see how the query module is meant to be used, but static variables might cause some problems, e.g.: reloading the query module might erase their value.

Reply:
I believe this is acceptable here.